prow/config/config.go

```diff
@@ -864,15 +870,23 @@ func mergeJobConfigs(a, b JobConfig) (JobConfig, error) {
 	return c, nil
 }
 
-func setPresubmitDecorationDefaults(c *Config, ps *Presubmit) {
+func setPresubmitDecorationDefaults(c *Config, ps *Presubmit, repo string) {
 	if ps.Decorate {
-		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		def := c.Plank.DefaultDecorationConfig
+		if dcByRepo, ok := c.Plank.DecorationConfigByRepo[repo]; ok {
+			def = dcByRepo.ApplyDefault(c.Plank.DefaultDecorationConfig)
+		}
+		ps.DecorationConfig = ps.DecorationConfig.ApplyDefault(def)
 	}
 }
 
-func setPostsubmitDecorationDefaults(c *Config, ps *Postsubmit) {
+func setPostsubmitDecorationDefaults(c *Config, ps *Postsubmit, repo string) {
```

Comment:
We should key off of `extra_refs[0]` for periodics, if set, to do this as well

Reply:
addressed